x-pack/plugins/security_solution/public/common/components/filter_group/filter_by_assignees.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { memo, useCallback, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { EuiFilterButton, EuiFilterGroup, EuiToolTip } from '@elastic/eui';
+
+import { TEST_IDS } from './constants';
+import { AssigneesPopover } from '../assignees/assignees_popover';
+import type { AssigneesIdsSelection } from '../assignees/types';
+import { useLicense } from '../../hooks/use_license';
+import { useUpsellingMessage } from '../../hooks/use_upselling';
+
+export interface FilterByAssigneesPopoverProps {
+  /**
+   * Ids of the users assigned to the alert
+   */
+  assignedUserIds: AssigneesIdsSelection[];
+
+  /**
+   * Callback to handle changing of the assignees selection
+   */
+  onSelectionChange?: (users: AssigneesIdsSelection[]) => void;
+}
+
+/**
+ * The popover to filter alerts by assigned users
+ */
+export const FilterByAssigneesPopover: FC<FilterByAssigneesPopoverProps> = memo(
+  ({ assignedUserIds, onSelectionChange }) => {
+    const isPlatinumPlus = useLicense().isPlatinumPlus();
+    const upsellingMessage = useUpsellingMessage('alert_assignments');
+
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+    const togglePopover = useCallback(() => setIsPopoverOpen((value) => !value), []);
+
+    const [selectedAssignees, setSelectedAssignees] =
+      useState<AssigneesIdsSelection[]>(assignedUserIds);
```

Comment:
Since `selectedAssignees` is only needed for the count of selectedAssignees, I think it would be easier to understand if this state held only the count. As it stands right now, it's hard to tell quickly which component truly owns the `selectedAssignee` state since `FilterByAssignessPopover`, `DetectionEnginePageComponent`, and `AssigneesApplyPanel` all keep a copy of the selection state.

Alternatively, this state could be passed in from the parent component rather than duplicating it here.

Reply:
As discussed, I will follow up on this one in a separate PR. Here is the [ticket](https://github.com/elastic/security-team/issues/8164).